includes/classes/order.php

```diff
@@ -854,43 +859,56 @@ function calculateTaxForProduct($index, $taxRates, $taxCountryId, $taxZoneId)
         //
         if (DISPLAY_PRICE_WITH_TAX === 'true') {
             $shown_price =
-                $currencies->value(zen_add_tax($product_final_price, $product_tax_rate)) * $product_qty
-                    + $currencies->value(zen_add_tax($product_onetime_charges, $product_tax_rate));
-
-            // -----
-            // Calculate the amount of tax included in the price when tax-in pricing is enabled.
-            //
-            $tax_add =
-                $currencies->value(zen_calculate_tax($product_final_price, $product_tax_rate)) * $product_qty
-                    + $currencies->value(zen_calculate_tax($product_onetime_charges, $product_tax_rate));
```

Comment:
Was the dropping of all calls to `$currencies->value()` (which does exchange rate conversion) intentional?

Reply:
Yes, since the order's base (as recorded in the database) values are kept in the site's DEFAULT_CURRENCY and converting them at this point caused various rounding errors.  The values are converted to the selected currency for display during customer and admin processing.

Reply:
While there's still one remaining issue, as posted by @piloujp on the forums [here](https://www.zen-cart.com/showthread.php?229752-Different-Total-Displayed&p=1399601#post1399601), my opinion is that these changes are **_much closer_** to full correction of the penny-off calculations that have been haunting the ZC base for years.

I'll work to address that issue (and possibly others) once this gets a bit wider testing via incorporation into the next release candidate.